src/Microsoft.ApplicationInsights.AspNet/TelemetryInitializers/SyntheticTelemetryInitializer.cs

```diff
@@ -0,0 +1,60 @@
+﻿namespace Microsoft.ApplicationInsights.AspNet.TelemetryInitializers
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Net;
+    using System.Net.Sockets;
+    using Extensibility.Implementation.Tracing;
+    using Microsoft.ApplicationInsights.Channel;
+    using Microsoft.ApplicationInsights.DataContracts;
+    using Microsoft.AspNet.Http;
+    using Microsoft.AspNet.Http.Features;
+
+    /// <summary>
+    /// This will allow to mark synthetic traffic from availability tests
+    /// </summary>
+    public class SyntheticTelemetryInitializer : TelemetryInitializerBase
+    {
+        private const string HeaderNameDefault = "GsmSyntheticTestRunId";
```

Comment:
Will remove that validation then and just validate if `telemetry.Context.Operation.SyntheticSource` is null or not.